presto-rcfile/src/main/java/com/facebook/presto/rcfile/HadoopCompressionCodec.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.rcfile;
+
+import io.airlift.slice.Slice;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.io.compress.CompressionInputStream;
+
+import java.io.IOException;
+
+import static java.util.Objects.requireNonNull;
+
+public class HadoopCompressionCodec
+        implements RcFileCompressionCodec
+{
+    private final CompressionCodec codec;
+
+    public HadoopCompressionCodec(CompressionCodec codec)
+    {
+        this.codec = requireNonNull(codec, "codec is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return codec.getClass().getName();
+    }
+
+    @Override
+    public void decompress(Slice compressed, Slice uncompressed)
+            throws IOException
+    {
+        CompressionInputStream decompressorStream = codec.createInputStream(compressed.getInput());
+        uncompressed.setBytes(0, decompressorStream, uncompressed.length());
+        decompressorStream.close();
```

Comment:
Use `try` because setBytes can throw if the decompressorStream produces less bytes than `uncompressed.length()`